vulnerabilities/importer.py

```diff
@@ -214,118 +213,38 @@ def __post_init__(self):
-            # arguments)
-            setattr(self.config, "last_run_date", last_run_date)
-            setattr(self.config, "cutoff_date", cutoff_date)
-        except Exception as e:
-            raise InvalidConfigurationError(str(e))
-
-        self.validate_configuration()
-
-    def __enter__(self):
-        pass
-
-    def __exit__(self, exc_type, exc_val, exc_tb):
-        pass
-
-    @classmethod
-    def qualified_name(cls):
-        """
-        Fully qualified name prefixed with the module name of the data source
-        used in logging.
-        """
-        return f"{cls.__module__}.{cls.__qualname__}"
-
-    @property
-    def cutoff_timestamp(self) -> int:
-        """
-        :return: An integer Unix timestamp of the last time this data source was queried or the
-        cutoff date passed in the constructor, whichever is more recent.
-        """
-        if not hasattr(self, "_cutoff_timestamp"):
-            last_run = 0
-            if self.config.last_run_date is not None:
-                last_run = int(self.config.last_run_date.timestamp())
+    spdx_license_expression = ""
 
-            cutoff = 0
-            if self.config.cutoff_date is not None:
-                cutoff = int(self.config.cutoff_date.timestamp())
+    def __init__(self):
+        if not self.spdx_license_expression:
+            raise Exception(f"Cannot run importer {self!r} without a license")
 
-            setattr(self, "_cutoff_timestamp", max(last_run, cutoff))
-
-        return self._cutoff_timestamp
-
-    def validate_configuration(self) -> None:
+    @classproperty
+    def qualified_name(self):
         """
-        Subclasses can perform more complex validation than what is handled by data classes and
-        their type annotations.
-
-        This method is called in the constructor. It should raise InvalidConfigurationError with a
-        human-readable message.
+        Fully qualified name prefixed with the module name of the improver used in logging.
         """
+        return f"{self.__module__}.{self.__qualname__}"
 
     def advisory_data(self) -> Iterable[AdvisoryData]:
         """
-        Subclasses return AdvisoryData objects
+        Return AdvisoryData objects corresponding to the data being imported
         """
         raise NotImplementedError
 
-    def error(self, msg: str) -> None:
-        """
-        Helper method for raising InvalidConfigurationError with the class name in the message.
-        """
-        raise InvalidConfigurationError(f"{type(self).__name__}: {msg}")
-
-
-@dataclasses.dataclass
-class GitDataSourceConfiguration(DataSourceConfiguration):
-    repository_url: str
-    branch: Optional[str] = None
-    create_working_directory: bool = True
-    remove_working_directory: bool = True
-    working_directory: Optional[str] = None
-
-
-class GitDataSource(DataSource):
-    CONFIG_CLASS = GitDataSourceConfiguration
 
+# TODO: Needs rewrite
```

Comment:
May be an issue to track may work better?

Reply:
This will be a part of https://github.com/nexB/vulnerablecode/issues/597